qmcpy/stopping_criterion/_cub_qmc_ld_g.py

```diff
@@ -156,3 +158,63 @@ def set_tolerance(self, abs_tol=None, rel_tol=None):
         """
         if abs_tol != None: self.abs_tol = abs_tol
         if rel_tol != None: self.rel_tol = rel_tol
+
+    def density_estimation(self, plot_estimation_flag=False):
+        approx_solution, data = self.integrate()
+        g=data.datum[0].y_val
+        if g.ndim != 1:
+            raise ValueError("The KDE only works for single-variate data.")
+        a=min(g)
+        b=max(g)
+        
+        unique_values, counts = unique(g, return_counts=True)
+        repeated_values = unique_values[counts > 1]# Find elements that are repeated (count > 1)
+        values_cont=unique_values[counts==1]
+        kde_discrete=counts[counts > 1]/len(g) #probability of each discrete value
+        prob=len(values_cont)/len(g) #probability of being continous
+        kde_cont=sc.stats.gaussian_kde(values_cont)
+
+        def kde_evaluate(values):
+            """
+            Args:
+            values: A vector of n points. 
+            """
+            values = atleast_1d(values)
+            if values.ndim != 1:
+                raise ParameterError("Values must be a vector of n points.")
+            densities = kde_cont.evaluate(values)*prob
+            discrete_indices = []
+            discrete_values = []
+            for i in range(len(values)):
+                if values[i] in repeated_values:
+                    densities[i] = kde_discrete
```

Comment:
`densities.ndim==1` yet we are assigning an element to `kde_discrete` which is a vector? I think you need to find the index `j` where `values[i]` is in in `repeated_values` then set `densities[i] = kde_discrete[j]`. Let me know if I'm misunderstanding